File: Python/SLP/SLP_Driver.py
import sys
import os
sys.path.append(os.path.join(os.path.dirname(__file__), "../../MNIST"))
from MNIST import MNIST_Image
from SLP import SLP

def main():
	trainingImages = MNIST_Image.getTrainingData()
	testingImages = MNIST_Image.getTestingData()
	for image in trainingImages:
		image.simplify()
	for image in testingImages:
		image.simplify()
	slp = SLP(10, 28*28, 0.07)
	slp.train(trainingImages)
	slp.test(testingImages)



if __name__=="__main__":
	main()

File: C/SLP/SLP_Driver.c
#include <stdio.h>
#include <stdlib.h>
#include <inttypes.h>
#include "../../MNIST/MNIST.h"
#include "./SLP.h"

/**
 * The driver for a Single Layer Perceptron that attempts to classify digit images.
 *
 */
int main(int argc, const char* argv)
{
	MNIST_Image* trainingImages = GetTrainingImages();
	MNIST_Image* testingImages = GetTestingImages();
	printf("Simplifying Training Images:\n");
	for (uint32_t imageIndex = 0; imageIndex < 60000; ++imageIndex)
	{
		SimplifyImage(&(*(trainingImages + imageIndex)));
		printf("%d/%d\r", imageIndex + 1, 60000);
	}
	printf("\nSimplifying Testing Images:\n");
	for (uint32_t imageIndex = 0; imageIndex < 10000; ++imageIndex)
	{
		SimplifyImage(&(*(testingImages + imageIndex)));
		printf("%d/%d\r", imageIndex + 1, 10000);
	}
	printf("\n");
	SLP slp = InitSLP(10, 0.07);
	Train(&slp, trainingImages, 60000);
	Test(&slp, testingImages, 10000);
	FreeSLP(slp);
	FreeImages(trainingImages, 60000);
	FreeImages(testingImages, 10000);
}